### PR TITLE
TRUNK-6598: Add missing unit test for containingAnyFormField in HibernateFormDAO

### PR DESCRIPTION
**Pull request by ShrutiSocrates**
TRUNK-6598: Add missing unit test for containingAnyFormField in HibernateFormDAO

## Description of what I changed
Added a missing unit test for the  containingAnyFormField functionality in HibernateFormDAO as indicated by the TODO comment on line 499.

## Issue I worked on
see https://issues.openmrs.org/browse/TRUNK-6598

## Checklist: I completed these to help reviewers :)
- [x] My IDE is configured to follow the code style
- [x] I have added tests to cover my changes
- [x] I ran mvn clean package
- [x] All new and existing tests passed
- [x] My pull request is based on latest master

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java [hunk 1]
```diff
@@ -64,4 +66,27 @@ public void shouldGetFormFieldsByForm() {
...
+			Collections.emptyList()         // fields
+		);
+
+		// Assert that forms were found
+		assertNotNull(forms);
+
+		// Assert at least one form contains this form field
+		assertTrue(forms.size() > 0);
+	}
 }
```
Comment (VarshithReddy2006) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:120]: Are there any edge cases where multiple form fields exist but none match? It might be useful to include such scenarios in the tests.

**Review by VarshithReddy2006 — COMMENTED** [review]
Suggested adding additional edge case coverage for test scenarios.

**Review by NitinKumar1-1 — COMMENTED** [review]
Looks good to me 
Agree with the suggestion about adding an edge case for non-matching form fields.

**Review by chaiwat123431 — COMMENTED** [review]
Good addition. Filling in a TODO with a meaningful test is exactly the kind of contribution that improves long-term maintainability. The test covers the happy path well. Consider also adding an edge case where no form fields match, to verify the method returns an empty result rather than throwing an exception.

**Comment by ShrutiSocrates** [comment]
Thank you for the feedback!

I have added an edge case to verify the method returns an empty result when no form fields match rather 
than throwing an exception.

Please review when you get a chance!

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java [hunk 2]
```diff
@@ -64,4 +66,43 @@ public void shouldGetFormFieldsByForm() {
...
+		List<FormField> invalidFields = Arrays.asList(new FormField(999));
+		List<Form> formsInvalid = dao.getForms(null, false,
+			Collections.emptyList(), null,
+			invalidFields, Collections.emptyList(),
+			Collections.emptyList());
+		assertNotNull(formsInvalid);
+		assertTrue(formsInvalid.isEmpty());
+	}
+
 }
```
Comment (EDSONZ-WASSWA) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:120]:
```suggestion

		
}

@Test
@@ -64,4 +66,43 @@
assertEquals(form.getFormId(), formField.getForm().getFormId());
}
}

	@Test
	public void getForms_shouldReturnFormsContainingAnyFormField() {
		// Test 1 - Basic: should return forms containing any form field
		List<FormField> anyFormFields = Arrays.asList(new FormField(2));
		List<Form> forms = dao.getForms(null, false,
			Collections.emptyList(), null,
			anyFormFields, Collections.emptyList(),
			Collections.emptyList());
		assertNotNull(forms);
		assertTrue(forms.size() > 0);

		// Test 2 - Empty list: should return all forms
		List<Form> formsWithEmpty = dao.getForms(null, false,
			Collections.emptyList(), null,
			Collections.emptyList(), Collections.emptyList(),
			Collections.emptyList());
		assertNotNull(formsWithEmpty);

		// Test 3 - Multiple fields: should return forms with any field
		List<FormField> multipleFields = Arrays.asList(
			new FormField(2), new FormField(3));
		List<Form> formsMultiple = dao.getForms(null, false,
			Collections.emptyList(), null,
			multipleFields, Collections.emptyList(),
			Collections.emptyList());
		assertNotNull(formsMultiple);
		assertTrue(formsMultiple.size() > 0);

		// Test 4 - Non existent field: should return empty list
		List<FormField> invalidFields = Arrays.asList(new FormField(999));
		List<Form> formsInvalid = dao.getForms(null, false,
			Collections.emptyList(), null,
			invalidFields, Collections.emptyList(),
			Collections.emptyList());
		assertNotNull(formsInvalid);
		assertTrue(formsInvalid.isEmpty());
	}

}
//A would-be fix for that
List<FormField> multipleInvalidFields = Arrays.asList(
	new FormField(998), 
	new FormField(999));
List<Form> formsMultipleInvalid = dao.getForms(null, false,
	Collections.emptyList(), null,
	multipleInvalidFields, Collections.emptyList(),
	Collections.emptyList());
assertNotNull(formsMultipleInvalid);
assertTrue(formsMultipleInvalid.isEmpty());
```

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java (hunk 1 above)
Comment (EDSONZ-WASSWA) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:120]: Good point. You're right—we should cover the scenario with multiple 
non-existent fields

Comment (ShrutiSocrates) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:120]: Thank you for the suggestion! I have committed it directly.

**Comment by ShrutiSocrates** [comment]
> Suggested adding additional edge case coverage for test scenarios.

Good point! I have added this edge case covering multiple non-existent form fields that return an empty list.

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java (hunk 1 above)
Comment (EDSONZ-WASSWA) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:120]: Nice work! @ShrutiSocrates .. I see there is thorough test coverage for the containingAnyFormField functionality as the ticket required....Looks good
Just one thing the comment "//A would-be fix for that" should be removed before merge

**Review by EDSONZ-WASSWA — APPROVED** [review]

**Comment by ShrutiSocrates** [comment]
Thank you so much @EDSONZ-WASSWA  for the approval and feedback!  I will remove the "//A would-be fix for that" comment right away.

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java (hunk 1 above)
Comment (ShrutiSocrates) [reply api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:120]: Thank you for the approval and feedback! The "//A would-be fix for that" comment 
has been removed.

**Review by jwnasambu — COMMENTED** [review]
Kindly restore the accidentally deleted assertion logic for this existing
     test for without the assertions the test is "empty" and provides no verification of the code’s correctness.

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java [hunk 3]
```diff
@@ -49,19 +50,81 @@ public void shouldFilterAgainstFormFields() {
...
 
+		
 	}
 
 	@Test
 	public void shouldGetFormFieldsByForm() {
 		Form form = new Form(2);
 		List<FormField> formFields = dao.getFormFields(form);
 
-		assertNotNull(formFields);
```
Comment (jwnasambu) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:59]: Kindly restore this accidentally deleted assertion logic for this existing
     test for without the assertions the test is "empty" and provides no verification of the code’s correctness.

File: api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java [hunk 4]
```diff
@@ -49,19 +50,81 @@ public void shouldFilterAgainstFormFields() {
...
+		
+}
+
+@Test
+@@ -64,4 +66,43 @@
```
Comment (jwnasambu) [inline api/src/test/java/org/openmrs/api/db/hibernate/HibernateFormDAOTest.java:65]: Kindly this is not a valid Java code (I believe it was likely introduced by a copy paste error from a git diff window and prevents the project from building) just delete that line
